### PR TITLE
Implemented trailing_zeros and leading_zeros

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

**Comment by debris** [comment]
LGTM

File: src/uint.rs [hunk 1]
```diff
@@ -622,6 +622,36 @@ macro_rules! construct_uint {
 				arr[index / 64] & (1 << (index % 64)) != 0
 			}
 
+			/// Returns the number of leading zeros in the binary representation of self.
+			pub fn leading_zeros(&self) -> u32 {
+				let mut r = 0;
+				for i in 0..$n_words {
+					let w = self.0[$n_words - i - 1];
```
Comment (rphmeier) [inline src/uint.rs:629]: `for w in self.0.iter().rev()`?

File: src/uint.rs [hunk 2]
```diff
@@ -622,6 +622,36 @@ macro_rules! construct_uint {
 				arr[index / 64] & (1 << (index % 64)) != 0
 			}
 
+			/// Returns the number of leading zeros in the binary representation of self.
+			pub fn leading_zeros(&self) -> u32 {
```
Comment (tomusdrw) [inline src/uint.rs:626]: There is already `bits` method that could be used to calculate that.

File: src/uint.rs [hunk 3]
```diff
@@ -622,6 +622,36 @@ macro_rules! construct_uint {
...
+						r += 64;
+					} else {
+						r += w.leading_zeros();
+						break;
+					}
+				}
+				r
+			}
+
+			/// Returns the number of leading zeros in the binary representation of self.
```
Comment (tomusdrw) [inline src/uint.rs:640]: Comment is invalid.

**Pull request merged**